### ISSUE TITLE
Are SIMD instructions enabled when the z library is compiled for the zscale filter?

### ISSUE THREAD

**Issue by acc987** [GyanD/codexffmpeg#52]
On my Intel system, it seems that the `zscale` filter is 4x slower than a combination of the `colorspace` and `scale` filters. This contradicts the performance claims on https://github.com/sekrit-twc/zimg, so I was wondering whether you could verify that the z library is compiled with SIMD instructions enabled?

Are you providing any information stating the version of third-party libraries that you are using for building FFmpeg? An outdated version of the library being used might also explain my observations.

Thanks in advance for your assistance!

**Comment by GyanD** [comment]
Can you share all the test commands?

Libraries are compiled from HEAD each Sunday 

Config is

```
checking whether C compiler accepts -mtune=sandybridge... yes
checking whether C compiler accepts -mtune=ivybridge... yes
checking whether C compiler accepts -mtune=haswell... yes
checking whether C compiler accepts -mtune=skylake-avx512... yes
checking whether C compiler accepts -mtune=cascadelake... yes
checking whether C compiler accepts -mavx512f -mavx512pf -mavx512er -mavx512cd -mavx512vl -mavx512bw -mavx512dq -mavx512ifma -mavx512vbmi -mavx512vbmi2 -mavx512bitalg -mavx512vpopcntdq -mavx512vnni... yes

#define ZIMG_X86 1
#define ZIMG_X86_AVX512 1
```

**Comment by acc987** [comment]
Somewhat synthetic example commands that reproduce the decreased performance (instead of the expected increase) on my system are as follows:

Using zscale:
```
ffmpeg -i Input.mp4 -vf "zscale=pin=bt709:tin=bt709:min=bt709:w='min(round(360*dar),640)':h='min(round(640/dar),360)':f=lanczos:p=smpte170m:t=smpte170m:m=smpte170m:r=tv,pad=640:360:-1:-1,setdar=dar=16/9" Output.mp4
```

Using colorspace,scale,colorspace:
```
ffmpeg -i Input.mp4 -vf "colorspace=iall=bt709:all=smpte170m:trc=linear,scale='min(round(360*dar),640)':'min(round(640/dar),360)':flags=lanczos+accurate_rnd,colorspace=all=smpte170m:range=tv,pad=640:360:-1:-1,setdar=dar=16/9" Output.mp4
```

**Comment by GyanD** [comment]
So ffmpeg did not add slice threading support for zscale till commit d0aefc3706 on Mar 3rd 2022. In newer builds, zscale is faster than swscale.
5.0 branch does not have that support.

**Comment by acc987** [comment]
Ah, much obliged! I am not that familiar with FFmpeg's branching structure but I take it that the threading support you mentioned is part of git master but not included in release/5.0? So if I use a git master build instead I should see an improvement?

**Comment by GyanD** [comment]
Yes, a recent git master build will show improvement.
